packages/core/src/DirectoryLoader.ts

```diff
@@ -168,6 +182,7 @@ export abstract class DirectoryLoader {
 			className: credentialName,
 			sourcePath: filePath,
 			extends: tempCredential.extends,
+			supportedNodes: this.nodesByCredential[tempCredential.name],
```

Comment:
Since this assumes that nodes will always be loaded before credentials, I think we should add a test for it.